### PR TITLE
Added stats for Armour and Physical Damage Reduction

### PR DESCRIPTION
**Pull request by bullgaria**
This continues/copies PR #336 
...I managed to do a wonderful job of screwing up my branch, so I'm resubmitting it...

### CODE REVIEW

File: Modules/CalcOffence-3_0.lua [hunk 1]
```diff
@@ -2225,8 +2236,12 @@ function calcs.offence(env, actor, activeSkill)
             local storedDamageMore = round(skillModList:More(cfg, "ImpaleEffect"), 2)
             local storedDamageModifier = (1 + storedDamageInc) * storedDamageMore
             local impaleStoredDamage = baseStoredDamage * storedDamageModifier
+            local impaleHitDamageMod = impaleStoredDamage * impaleStacks  -- Source: https://www.reddit.com/r/pathofexile/comments/chgqqt/impale_and_armor_interaction/
+
+            local impaleArmourReduction = calcs.armourReductionF(round(calcLib.val(enemyDB, "Armour")), impaleHitDamageMod * output.impaleStoredHitAvg)
+            local impaleResist = m_max(0, enemyDB:Sum("BASE", nil, "PhysicalDamageReduction") + skillModList:Sum("BASE", cfg, "EnemyPhysicalDamageReduction") + skillModList:Sum("BASE", cfg, "EnemyImpalePhysicalDamageReduction") + impaleArmourReduction)
```
Comment (LocalIdentity) [inline Modules/CalcOffence-3_0.lua:2242]: The Physical Damage Reduction for Impales should be shown in the Impale multiplier breakdown. Can add something like this to line 2263 `t_insert(breakdown.ImpaleModifier, s_format("x %.2f ^8(impale physical damage reduction)", (1 - impaleResist / 100)))`

File: Modules/CalcOffence-3_0.lua [hunk 2]
```diff
@@ -1107,14 +1108,28 @@ function calcs.offence(env, actor, activeSkill)
...
-							resist = m_max(0, enemyDB:Sum("BASE", nil, "PhysicalDamageReduction") + skillModList:Sum("BASE", cfg, "EnemyPhysicalDamageReduction"))
+                            if isAttack then
+								-- store pre-armour physical damage from attacks for impale calculations
+								if pass == 1 then
+									output.impaleStoredHitAvg = output.impaleStoredHitAvg + damageTypeHitAvg * (output.CritChance / 100)
+								else
+									output.impaleStoredHitAvg = output.impaleStoredHitAvg + damageTypeHitAvg * (1 - output.CritChance / 100)
+								end
+							end
+							local armourReduction = calcs.armourReductionF(round(calcLib.val(enemyDB, "Armour")), damageTypeHitAvg)
```
Comment (LocalIdentity) [inline Modules/CalcOffence-3_0.lua:1131]: The armour reduction should take into account the 33% armour increase on Shaper or the 66% increase for Sirus when you select shaper/guardian on the configs page

File: Modules/CalcOffence-3_0.lua [hunk 3]
```diff
@@ -2225,8 +2236,12 @@ function calcs.offence(env, actor, activeSkill)
             local storedDamageMore = round(skillModList:More(cfg, "ImpaleEffect"), 2)
             local storedDamageModifier = (1 + storedDamageInc) * storedDamageMore
             local impaleStoredDamage = baseStoredDamage * storedDamageModifier
+            local impaleHitDamageMod = impaleStoredDamage * impaleStacks  -- Source: https://www.reddit.com/r/pathofexile/comments/chgqqt/impale_and_armor_interaction/
+
+            local impaleArmourReduction = calcs.armourReductionF(round(calcLib.val(enemyDB, "Armour")), impaleHitDamageMod * output.impaleStoredHitAvg)
```
Comment (LocalIdentity) [inline Modules/CalcOffence-3_0.lua:2241]: Should also take into account the armour increase for Shaper/Sirus

File: Modules/CalcOffence-3_0.lua (hunk 2 above)
Comment (bullgaria) [reply Modules/CalcOffence-3_0.lua]: I set them to 33% for now...it fits the most enemies' stats this way.
The 33% doesn't apply to: Minotaur = 100%, Phoenix = 50%, Hydra = 0%, Chimera = 0%

We really need Sirus in the dropdown, but that's not for this PR.

File: Modules/CalcOffence-3_0.lua (hunk 1 above)
Comment (bullgaria) [reply Modules/CalcOffence-3_0.lua:2286]: "impale enemy physical damage reduction"...too wordy?

File: Modules/CalcDefence-3_0.lua [hunk 4]
```diff
@@ -29,9 +29,14 @@ function calcs.hitChance(evasion, accuracy)
...
--- Calculate physical damage reduction from armour
+-- Calculate physical damage reduction from armour, float
+function calcs.armourReductionF(armour, raw)
+	return (armour / (armour + raw * 10) * 100)
+end
+
+-- Calculate physical damage reduction from armour, int
 function calcs.armourReduction(armour, raw)
-	return round(armour / (armour + raw * 10) * 100)
+	return round(calcs.armourReductionF(armour, raw))
```
Comment (LocalIdentity) [inline Modules/CalcDefence-3_0.lua:39]: Is there a reason this round function is only used for player damage reduction but not enemy damage reduction?

**Comment by LocalIdentity** [comment]
You'll also need to rebase the PR to keep it up to date with the current dev branch

File: Modules/CalcDefence-3_0.lua (hunk 4 above)
Comment (bullgaria) [reply Modules/CalcDefence-3_0.lua:39]: The original damage reduction was rounded, and really only used for display purposes since no further calculations are being done with player armour.
I kept it there to avoid breaking any existing functionality.

**Comment by bullgaria** [comment]
> You'll also need to rebase the PR to keep it up to date with the current dev branch

Yeah, did that just now...should be good to go.
